docs/modules/plugins/pages/plugin-json.adoc

```diff
@@ -182,3 +182,138 @@ When I issue a HTTP GET request for a resource with the URL 'http://jsonpath.her
+* Restores previously set context
+* Step will fail if variable never be set along the iterations
+
+.JSON based step
+[source,gherkin]
+----
+When I find $comparisonRule `$elementsNumber` JSON elements from `$json` by `$jsonPath` and until variable `$variableName` $variableMatcher `$expectedValue` for each element do:$stepsToExecute
+----
+
+.JSON based step alias
+[source,gherkin]
+----
+When I find $comparisonRule '$elementsNumber' JSON elements from '$json' by '$jsonPath' and until variable '$variableName' $variableMatcher '$expectedValue' for each element do:$stepsToExecute
+----
+
+
+* `comparisonRule` - xref:parameters:comparison-rule.adoc[comparison rule]
+* `elementsNumber` - expected number of elements
+* `json` - JSON data
+* `jsonPath` - {json-path}
+* `variableName` - The name of variable to validate
+* `variableMatcher` - xref:parameters:string-comparison-rule.adoc[The string comparison rule]
+* `expectedValue` - The expected value of the variable
+* `stepsToExecute` - steps to perform on JSON elements
+
+.Context based step
+[source,gherkin]
+----
+When I find $comparisonRule `$elementsNumber` JSON elements from context by `$jsonPath` and until variable `$variableName` $variableMatcher `$expectedValue` for each element do:$stepsToExecute
+----
+
+.Context based step alias
+[source,gherkin]
+----
+When I find $comparisonRule '$elementsNumber' JSON elements from context by '$jsonPath' and until variable '$variableName' $variableMatcher '$expectedValue' for each element do:$stepsToExecute
+----
+
+* `comparisonRule` - xref:parameters:comparison-rule.adoc[comparison rule]
+* `elementsNumber` - expected number of elements
+* `jsonPath` - {json-path}
+* `variableName` - The name of variable to validate
+* `variableMatcher` - xref:parameters:string-comparison-rule.adoc[The string comparison rule]
+* `expectedValue` - The expected value of the variable
+* `stepsToExecute` - steps to perform on JSON elements
+
+.Find the title
+[source,gherkin]
+----
+When I execute HTTP GET request for resource with URL `http://jsonpath.herokuapp.com/json/goessner.json`
+When I find > `1` JSON elements from `${response}` by `$.store.book` and until variable `title` matches `M.+` for each element do:
+|step|
+|When I save JSON element value from context by JSON path `$.title` to scenario variable `title`|
+Then `Moby Dick` is = `${title}`
+When I execute HTTP GET request for resource with URL `http://jsonpath.herokuapp.com/json/goessner.json`
+When I find > `1` JSON elements from context by `$.store.book` and until variable `title` matches `S.+` for each element do:
+|step|                                                                                                                     
+|When I save JSON element value from context by JSON path `$.title` to scenario variable `title`|                          
```

Comment:
similarly to https://github.com/vividus-framework/vividus/pull/2623/files#r840257037